src/openloco/interop/interop.hpp

```diff
@@ -130,58 +130,180 @@ namespace openloco::interop
     int32_t call(int32_t address);
     int32_t call(int32_t address, registers& registers);
 
-    template<typename T, uint32_t TAddress>
+    template<typename T, uintptr_t TAddress>
     struct loco_global
     {
-        void operator=(T rhs)
+        typedef T type;
+        typedef type* pointer;
+        typedef type& reference;
+        typedef const type& const_reference;
+
+        operator reference()
         {
-            addr<TAddress, T>() = rhs;
+            return addr<TAddress, T>();
         }
 
-        operator T&()
+        loco_global& operator=(const_reference v)
         {
-            return addr<TAddress, T>();
+            addr<TAddress, T>() = v;
+            return *this;
         }
 
-        T& operator*()
+        loco_global& operator+=(const_reference v)
+        {
+            addr<TAddress, T>() += v;
+            return *this;
+        }
+
+        loco_global& operator|=(const_reference v)
+        {
+            addr<TAddress, T>() |= v;
+            return *this;
+        }
+
+        loco_global& operator&=(const_reference v)
+        {
+            addr<TAddress, T>() &= v;
+            return *this;
+        }
+
+        loco_global& operator-=(const_reference v)
+        {
+            addr<TAddress, T>() -= v;
+            return *this;
+        }
+
+        loco_global& operator++()
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global operator++(int)
+        {
+            addr<TAddress, T>()++;
+            return *this;
+        }
+
+        loco_global& operator--()
+        {
+            addr<TAddress, T>()--;
+            return *this;
+        }
+
+        loco_global operator--(int)
+        {
+            addr<TAddress, T>()--;
+            return *this;
```

Comment:
`loco_global::operator--(int)` is wrong in exactly the same manner as `loco_global::operator++(int)` and should be corrected in the same manner. (See comment above.)